src/mod_filtering.c

```diff
@@ -125,8 +131,19 @@ static void audio_frame_to_sound_texture(struct filtering_ctx *ctx, AVFrame *dst
          * The imaginary parts for these two frequencies are always 0 so they
          * are assumed as such. This trick allowed an in-place processing for
          * the N samples into N+1 complex.
+         *
+         * After avtx_func(), the bins are an array of successive real and
+         * imaginary floats of size NB_SAMPLES/2 + 1. To mimic the
+         * av_rdft_calc() behavior, we simply copy the real part of the higher
+         * frequency into the imaginary part of the first bin. It also requires
+         * the bins allocation to be at least of size (nb_samples + 2) * sizeof(float).
          */
+#if LIBAVUTIL_VERSION_INT >= AV_VERSION_INT(57, 19, 100)
+        ctx->avtx_func(ctx->avtx, bins, bins, sizeof(float));
+        bins[1] = bins[nb_samples];
```

Comment:
Note that this should be unnecessary since we drop the highest frequency (this bin is ignored, see `fft_dst[0] = ...` below)

Reply:
I kept it to make it consistent with the rdft codepath. I think, it will make sense to remove it when we remove the rdft codepaths.